spec/unit/webrtc/groupCall.spec.ts

```diff
@@ -280,6 +289,143 @@ describe('Group Call', function() {
+                const call = new MockCall(room.roomId, groupCall.groupCallId);
+                call.getOpponentMember = (() => {}) as unknown as any;
+
+                // @ts-ignore Mock
+                expect(() => groupCall.onCallFeedsChanged(call)).toThrowError();
+            });
+
+            describe("usermedia feeds", () => {
+                it("adds new usermedia feed", async () => {
+                    // @ts-ignore Mock
+                    call.remoteUsermediaFeed = newFeed;
+                    // @ts-ignore Mock
+                    groupCall.onCallFeedsChanged(call);
+
+                    // @ts-ignore Mock
+                    expect(groupCall.addUserMediaFeed).toHaveBeenCalledWith(newFeed);
+                });
+
+                it("replaces usermedia feed", async () => {
+                    groupCall.userMediaFeeds = [currentFeed.typed()];
+
+                    // @ts-ignore Mock
+                    call.remoteUsermediaFeed = newFeed;
+                    // @ts-ignore Mock
+                    groupCall.onCallFeedsChanged(call);
+
+                    // @ts-ignore Mock
+                    expect(groupCall.replaceUserMediaFeed).toHaveBeenCalledWith(currentFeed, newFeed);
+                });
+
+                it("removes usermedia feed", async () => {
+                    groupCall.userMediaFeeds = [currentFeed.typed()];
+
+                    // @ts-ignore Mock
+                    groupCall.onCallFeedsChanged(call);
+
+                    // @ts-ignore Mock
+                    expect(groupCall.removeUserMediaFeed).toHaveBeenCalledWith(currentFeed);
+                });
+            });
+
+            describe("screenshare feeds", () => {
+                it("adds new screenshare feed", async () => {
+                    // @ts-ignore Mock
+                    call.remoteScreensharingFeed = newFeed;
+                    // @ts-ignore Mock
+                    groupCall.onCallFeedsChanged(call);
+
+                    // @ts-ignore Mock
+                    expect(groupCall.addScreenshareFeed).toHaveBeenCalledWith(newFeed);
+                });
+
+                it("replaces screenshare feed", async () => {
+                    groupCall.screenshareFeeds = [currentFeed.typed()];
+
+                    // @ts-ignore Mock
+                    call.remoteScreensharingFeed = newFeed;
+                    // @ts-ignore Mock
+                    groupCall.onCallFeedsChanged(call);
+
+                    // @ts-ignore Mock
+                    expect(groupCall.replaceScreenshareFeed).toHaveBeenCalledWith(currentFeed, newFeed);
+                });
+
+                it("removes screenshare feed", async () => {
+                    groupCall.screenshareFeeds = [currentFeed.typed()];
+
+                    // @ts-ignore Mock
+                    groupCall.onCallFeedsChanged(call);
+
+                    // @ts-ignore Mock
+                    expect(groupCall.removeScreenshareFeed).toHaveBeenCalledWith(currentFeed);
```

Comment:
Same goes for this and the 5 or so tests above - is there any reasonable way to test this without relying on internals?